codec/src/main/java/io/netty/handler/codec/compression/LzmaFrameEncoder.java

```diff
@@ -212,4 +214,29 @@ private static int maxOutputBufferLength(int inputLength) {
         }
         return 13 + (int) (inputLength * factor);
     }
+
+    @Override
+    public boolean isClosed() {
+        return false;
+    }
+
+    @Override
+    public ChannelFuture close() {
+        return ctx().newFailedFuture(new UnsupportedOperationException());
+    }
+
+    @Override
+    public ChannelFuture close(ChannelPromise promise) {
+        return promise.setFailure(new UnsupportedOperationException());
+    }
+
+    @Override
+    public void close(ChannelHandlerContext ctx, ChannelPromise promise) throws Exception {
+        ctx.close(promise);
+    }
+
+    @Override
+    protected ChannelFuture finishEncode(ChannelHandlerContext ctx, ChannelPromise promise) {
+        return promise.setFailure(new UnsupportedOperationException());
```

Comment:
Instead of rejecting the close request, we could simply close the channel immediately without scheduling any write timeout, etc?